overlord/snapstate/autorefresh_test.go

```diff
@@ -1002,10 +1004,35 @@ func (s *autoRefreshTestSuite) TestInhibitRefreshRefreshesWhenOverdue(c *C) {
 		Sequence:             []*snap.SideInfo{si},
 		Current:              si.Revision,
 		RefreshInhibitedTime: &pastInstant,
+		Flags:                snapstate.Flags{IsAutoRefresh: true},
 	}
 	err := snapstate.InhibitRefresh(s.state, snapst, info, func(si *snap.Info) error {
 		return &snapstate.BusySnapError{SnapInfo: si}
 	})
 	c.Assert(err, IsNil)
 	c.Check(notificationCount, Equals, 1)
 }
+
+func (s *autoRefreshTestSuite) TestInhibitNoNotificationOnManualRefresh(c *C) {
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	restore := snapstate.MockAsyncPendingRefreshNotification(func(ctx context.Context, client *userclient.Client, refreshInfo *userclient.PendingSnapRefreshInfo) {
+		c.Fatal("shouldn't trigger pending refresh notification if refresh was manual")
+	})
+	defer restore()
+
+	pastInstant := time.Now().Add(-snapstate.MaxInhibition)
+
+	si := &snap.SideInfo{RealName: "pkg", Revision: snap.R(1)}
+	info := &snap.Info{SideInfo: *si}
+	snapst := &snapstate.SnapState{
+		Sequence:             []*snap.SideInfo{si},
+		Current:              si.Revision,
+		RefreshInhibitedTime: &pastInstant,
```

Comment:
(nitpick^3) maybe include the explicit flag: `Flags:                snapstate.Flags{IsAutoRefresh: false},` or a comment like: `// Note that Flags does not include "IsAutoRefresh: true" which indicates a manual refresh`. But given the surrounding tests I may overthink this.